lib/pacemaker/pcmk_sched_bundle.c

```diff
@@ -575,6 +575,7 @@ pcmk__bundle_action_flags(pe_action_t *action, const pe_node_t *node)
 
     bundled_resource = pe__bundled_resource(action->rsc);
     if (bundled_resource != NULL) {
+        // Clone actions are done on the bundled clone resource, not container
```

Comment:
Still a bit surprising that start (for example) would be done on the container when it's applicable to a cloned resource as well. (Obviously promote has to run on the clone.) Maybe just another confusing aspect of how bundles work.

Reply:
Any action on one of the implicit resources will use the primitive or clone method as appropriate. This is for (pseudo-)actions on the bundle itself, e.g. "start bundle" or "promote bundle".

Reply:
I think that makes sense. Based on this, "start bundle" pseudo action =~ "start its containers" and "promote bundle" pseudo action =~ "promote its child (the clone)" at least for action flags purposes.